dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.ts

```diff
@@ -33,12 +33,23 @@ export const handleFeeRecordGenerateKeyingDataEvent = async (
     return await transactionEntityManager.save(FeeRecordEntity, feeRecord);
   }
 
-  const { coverPercentage } = await getLatestTfmFacilityValues(feeRecord.facilityId, reportPeriod);
-
-  const fixedFeeAdjustment = await calculateFixedFeeAdjustment(feeRecord, feeRecord.facilityUtilisationData, reportPeriod);
+  const { coverPercentage, coverEndDate, interestPercentage, dayCountBasis } = await getKeyingSheetCalculationTfmFacilityValues(
+    feeRecord.facilityId,
+    reportPeriod,
+  );
 
   const ukefShareOfUtilisation = calculateUkefShareOfUtilisation(feeRecord.facilityUtilisation, coverPercentage);
 
+  const fixedFee = calculateFixedFee({
+    ukefShareOfUtilisation,
+    reportPeriod,
+    coverEndDate,
+    interestPercentage,
+    dayCountBasis,
+  });
+
+  const fixedFeeAdjustment = calculateFixedFeeAdjustment(feeRecord, feeRecord.facilityUtilisationData, reportPeriod, fixedFee);
+
   const principalBalanceAdjustment = calculatePrincipalBalanceAdjustment(ukefShareOfUtilisation, feeRecord.facilityUtilisationData);
 
   const statusToUpdateTo = getStatusToUpdateTo(feeRecord.feesPaidToUkefForThePeriod, fixedFeeAdjustment, principalBalanceAdjustment);
```

Comment:
There is **a lot** going on in this function now - There are 10 calls and it is not possible to understand what it's doing from the documentation.

Happy to improve documentation in another PR.

Reply:
I think it might need another PR.  I am not fully sure what all the functions do, and it will look weird if I add documetnation for just the ones I know/have moved